yt_dlp/downloader/youtube_dl_from_start.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import unicode_literals
+
+import re
+from time import sleep
+
+from .dash import DashSegmentsFD
+from ..extractor.youtube import YoutubeIE
+
+from ..utils import (
+    traverse_obj,
+    urljoin,
+    time_millis,
+)
+
+
+class YoutubeDlFromStartDashFD(DashSegmentsFD):
+    """
+    Download YouTube live from the start, to the end. For DASH formats.
+    """
+
+    FD_NAME = 'ytlivestartdash'
+
+    @staticmethod
+    def _manifest_fragments(ie: YoutubeIE, mpd_url, stream_number, begin_index=0, fetch_span=5000):
+        known_idx = begin_index + 1
+        no_fragment_count = 0
+        prev_dl = time_millis()
+        while True:
+            if no_fragment_count > 5:
+                return
```

Comment:
Is this reliable? Maybe we should let the user press ctrl+c to end stream

Reply:
this is for when the downloader can't find no new fragments. Ctrl+C handling is the responsible of FragmentFD, not here

Reply:
tl;dr unreliable. requires to get header of **any segment**
https://github.com/Kethsar/ytarchive/blob/master/ytarchive.py#L1090